controllers/user.js

```diff
@@ -53,6 +53,77 @@ router.get( '/:id/lexemes', (req, res, next) => {
 });
 
 
+router.put( '/:id/lexemes/add/:lexemeId', (req, res, next) => {
+
+  const lexemeIdToAdd = req.params.lexemeId;
+
+  User.findById( req.user._id )
+    .then( user => {
+      const lexeme = user.lexemes.find( ({ lexeme: id }) => String(id) === lexemeIdToAdd );
+
+      if(lexeme) {
+        const err = new Error( 'Lexeme is already in learning list.' );
+        err.status = 409;
+        return next(err);
+      }
+
+      if(!lexeme) {
+        req.user.update( { lexemes: [...user.lexemes, { lexeme: lexemeIdToAdd }] }, (error, result) => {
+          if ( error ) return next(error);
+
+          let savedLexeme = result.lexemes.find( ({ lexeme: id }) => String(id) === lexemeIdToAdd );
+
+          Lexeme.findById( lexemeIdToAdd )
+            .then( foundLexeme => {
+              const returnLexeme = {
+                lexeme: {
+                  lexeme: foundLexeme.lexeme,
+                  _id: foundLexeme._id },
+                _id: savedLexeme._id,
+                progress: 1,
+                wrongAnswers: 0,
+                correctAnswers: 0
+                };
+
+              res.json(returnLexeme)
+            })
+            .catch( error => next(error) );
+        });
+      }
+    })
+    .catch( error => next(error) );
+});
+
+
+router.put( '/:id/lexemes/remove/:lexemeId', (req, res, next) => {
+
+  const lexemeIdToRemove = req.params.lexemeId;
+
+  User.findById( req.user._id )
+    .then( user => {
+      const lexeme = user.lexemes.find( ({ lexeme: id }) => String(id) === lexemeIdToRemove );
+
+      if(!lexeme) {
+        const err = new Error( 'Lexeme is not in learning list.' );
+        err.status = 409;
+        return next(err);
+      }
+
+      if(lexeme) {
+        const lexemesFiltered = user.lexemes
+          .filter( ({ lexeme: lexemeId}) => lexemeIdToRemove.indexOf(String(lexemeId)) === -1 );
```

Comment:
Why not just comparing?? `lexemeIdToRemove` is just one right? so why `indexOf`??  and not just `lexemeIdToRemove !== String(lexemeId)`